### ISSUE TITLE
SSL Certificate Error when downloading pretrained weights

### ISSUE THREAD

**Issue by grudloff** [pytorch/pytorch#109080]
### 🐛 Describe the bug

Just did a fresh install of windows 10, and installed vs-code, miniconda and git.

I used an environment created with the following enviroment.yml file:

```yml
name : ds_gpu
channels:
  - pytorch
  - nvidia
  - conda-forge
dependencies:
  - pytorch 
  - torchvision
  - torchaudio
  - pytorch-cuda=11.8
  - pytorch-lightning
  - matplotlib
  - seaborn
  - numpy
  - scipy
  - scikit-learn
  - jupyter
  - ipykernel
  - pandas
```

I am trying to train a model that uses a pretrained weight that is on: https://download.pytorch.org/models/efficientnet_v2_s-dd5fe13b.pth

But it fails with the following error

`URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1006)>`

Is this an issue on my side? Can someone else reproduce this?

### Minimum code to reproduce

```python
import torch

url = "https://download.pytorch.org/models/efficientnet_v2_s-dd5fe13b.pth"
model_weights = torch.hub.load_state_dict_from_url(url)
```

### Versions

Collecting environment information...
PyTorch version: 2.0.1
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Home
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.11.5 | packaged by conda-forge | (main, Aug 27 2023, 03:23:48) [MSC v.1936 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19045-SP0
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080
Nvidia driver version: 531.58
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture=9
CurrentClockSpeed=3200
DeviceID=CPU0
Family=107
L2CacheSize=3072
L2CacheSpeed=
Manufacturer=AuthenticAMD
MaxClockSpeed=3200
Name=AMD Ryzen 5 1600 Six-Core Processor
ProcessorType=3
Revision=257

Versions of relevant libraries:
[pip3] numpy==1.25.2
[pip3] pytorch-lightning==2.0.8
[pip3] torch==2.0.1
[pip3] torchaudio==2.0.2
[pip3] torchmetrics==1.1.2
[pip3] torchvision==0.15.2
[conda] No relevant packages

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @vladimir-aubrecht @iremyux @Blackhex @cristianPanaite @nairbv @NicolasHug @vmoens @jdsgomes @penguinwu

**Comment by janeyx99** [comment]
I cannot repro on my linux remote machine--it downloads fine. It most likely has to do with Windows/your local setup/perhaps both?

**Comment by iremyux** [comment]
@grudloff , 
I tried to reproduce this problem but couldn't with the same versions. I created the conda environment with the yml file you provided. Although I have Windows 11, it should not be a problem. 

Would you like to give more details on how to reproduce?

**Comment by iremyux** [comment]
I have just realized I used anaconda and not miniconda. So I have installed the latest version of miniconda for Windows and created a brand new environment with your yml file. The download completed succesfully.

**Comment by grudloff** [comment]
I managed to fix it, but I am still super confused about it. I created a "ds_test" environment:

```bash
conda create --name ds_test pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia
```

and somehow this is fixing the certificate issue.

I was very confused because I didn't make any other change, so I deleted both environments and followed all my steps and observed the same outcome.
1. Create environment through yml file.
2. Activate ds_gpu and try downloading weights -> SSL error
3. Create ds_test environment.
4. Activate ds_test and try downloading weights -> success
5. Delete weights
6. Activate ds_gpu and try downloading weights -> success

Looks like probably this is a conda issue?
